contracts/axelarnet-gateway/src/clients/external.rs

```diff
@@ -11,39 +12,47 @@ pub struct AxelarExecutableMsg {
     pub payload: HexBinary,
 }
 
-/// Crate-specific `ExecuteMsg` type wraps the `AxelarExecutableMsg` for the AxelarExecutable client.
+/// By convention, amplifier-compatible contracts must expose this `Execute` variant.
+/// Due to identical json serialization, we can imitate it here so the gateway can call it.
 #[cw_serde]
-pub enum AxelarExecutableExecuteMsg {
+pub enum ExecuteMsg {
     /// Execute the message at the destination contract with the corresponding payload.
     Execute(AxelarExecutableMsg),
 }
 
-impl<'a> From<client::Client<'a, AxelarExecutableExecuteMsg, ()>> for AxelarExecutableClient<'a> {
-    fn from(client: client::Client<'a, AxelarExecutableExecuteMsg, ()>) -> Self {
-        AxelarExecutableClient { client }
-    }
+pub struct CrossChainExecutor<'a> {
+    client: client::Client<'a, ExecuteMsg, ()>,
 }
 
-pub struct AxelarExecutableClient<'a> {
-    client: client::Client<'a, AxelarExecutableExecuteMsg, ()>,
-}
+impl<'a> CrossChainExecutor<'a> {
+    pub fn new(deps: Deps<'a>, destination: &str) -> error_stack::Result<Self, address::Error> {
+        let destination = address::validate_cosmwasm_address(deps.api, destination)?;
+        Ok(CrossChainExecutor {
+            client: client::Client::new(deps.querier, destination),
+        })
+    }
 
-impl<'a> AxelarExecutableClient<'a> {
-    pub fn execute(
+    pub fn prepare_execute_msg(
```

Comment:
```suggestion
    pub fn execute(
```
We name the client methods by the msg that they create. Should be consistent. Alternatively, don't create a new client struct, and define a single method `prepare_execute_msg` that constructs the client `client::Client<'a, AxelarExecutableExecuteMsg, ()>` and returns the msg in one-shot